examples/OpenGL45/matrix_multiplication/mat_mul4x4_shared_memory.shadertrap

```diff
@@ -0,0 +1,203 @@
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+    float 512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array1 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array2 BINDING 1
+BIND_SHADER_STORAGE_BUFFER BUFFER result BINDING 2
+
+DECLARE_SHADER shader KIND COMPUTE
+#version 450
+
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+
+layout(local_size_x=4, local_size_y=4, local_size_z=1) in;
+
+layout(binding = 0) buffer array1 {
+  float[16][16] data1;
+};
+layout(binding=1) buffer array2 {
+  float[16][16] data2;
+};
+layout(binding=2) buffer result{
+  float[16][16] res;
+};
+
+
+
+float[KERNEL_SIZE][KERNEL_SIZE] submatrix_mul(float[KERNEL_SIZE][KERNEL_SIZE] a,float[KERNEL_SIZE][KERNEL_SIZE] b){
+    float[KERNEL_SIZE][KERNEL_SIZE] tmp_buf;
+
+    int i,j,k;
+    for(i=0;i<KERNEL_SIZE;i++){
+        for(j=0;j<KERNEL_SIZE;j++)
+        tmp_buf[i][j] = 0;
+    }
+
+    for(i=0;i<KERNEL_SIZE;i++){
+        for(j=0;j<KERNEL_SIZE;j++){
+            for(k=0;k<KERNEL_SIZE;k++){
+                tmp_buf[i][j] += (a[i][k] * b[k][j]);
+            }
+        }
+    }
+    return tmp_buf;
+}
+
+float[KERNEL_SIZE][KERNEL_SIZE] take_submatrix(float[MAT_SIZE][MAT_SIZE] mat,uint x_a, uint y_a){
+    
+    float[KERNEL_SIZE][KERNEL_SIZE] res_a;
+    int i,j;
+
+    for(i=0;i<KERNEL_SIZE; i++){
+        for(j=0;j< KERNEL_SIZE;j++){
+            res_a[i][j] = mat[x_a*KERNEL_SIZE+i][y_a * KERNEL_SIZE + j];
+
+        }
+    }
+    return res_a;
```

Comment:
I recommend following Google C++-like rules for whitespace. For instance, a space after `;`in a for loop and a space either side of `<`.

I'd also remove the whitespace at the end of the inner loop as it's not consistent with the rest of your code.